lib/spack/spack/test/data/compilers.yaml

```diff
@@ -7,6 +7,10 @@ compilers:
       cxx: /path/to/clang++
       f77: None
       fc: None
+    implicit_link_paths:
+      - /lib
+      - /usr/lib
+      - /usr/lib/gcc/platform/version
```

Comment:
There needs to be a way here to avoid putting system paths like `/lib` in the implicit paths.  We don't want to add any RPATHs for system paths. 

One thing that is a little hazy here is where to draw the line between system paths and compiler runtime paths. *Probably* we should exclude everything that appears in `/etc/ld.so.conf`.  I would say that the main things to verify are a) that we can use intel compilers without Intel's setup script -- i.e., that we include RPATHs for Intel's runtime directories, and b) that we can use newer versions of gcc than the system version when the `libstdc++` implementations are incompatible.  i.e., gcc 4.9.3+ on RHEL6.